Content.Server/Chat/Managers/ChatManager.cs

```diff
@@ -138,68 +133,57 @@ public void EntitySay(EntityUid source, string message, bool hideChat=false)
-                    continue;
+            message = SanitizeMessageCapital(source, message);
 
-                clients.Add(player.ConnectedClient);
-            }
+            var listeners = EntitySystem.Get<ListeningSystem>();
+            listeners.PingListeners(source, message);
 
-            if (message.StartsWith(';'))
-            {
-                // Remove semicolon
-                message = message.Substring(1).TrimStart();
+            message = FormattedMessage.EscapeText(message);
 
-                // Capitalize first letter
-                message = message[0].ToString().ToUpper() +
-                          message.Remove(0, 1);
+            var messageWrap = Loc.GetString("chat-manager-entity-say-wrap-message",("entityName", Name: _entManager.GetComponent<MetaDataComponent>(source).EntityName));
+            NetMessageToMany(ChatChannel.Local, message, messageWrap, source, hideChat, clients);
+        }
 
-                var invSystem = EntitySystem.Get<InventorySystem>();
+        public void EntityWhisper(EntityUid source, string message, bool hideChat=false)
+        {
+            if (!EntitySystem.Get<ActionBlockerSystem>().CanSpeak(source))
+            {
+                return;
+            }
 
-                if (invSystem.TryGetSlotEntity(source, "ears", out var entityUid) &&
-                    _entManager.TryGetComponent(entityUid, out HeadsetComponent? headset))
-                {
-                    headset.RadioRequested = true;
-                }
-                else
-                {
-                    source.PopupMessage(Loc.GetString("chat-manager-no-headset-on-message"));
-                }
+            if (MessageCharacterLimit(source, message))
+            {
+                return;
             }
-            else
+
+            foreach (var handler in _chatTransformHandlers)
             {
-                // Capitalize first letter
-                message = message[0].ToString().ToUpper() +
-                          message.Remove(0, 1);
+                //TODO: rather return a bool and use a out var?
+                message = handler(source, message);
             }
 
+            message = message.Trim();
+
+            message = SanitizeMessageCapital(source, message);
+
             var listeners = EntitySystem.Get<ListeningSystem>();
             listeners.PingListeners(source, message);
 
             message = FormattedMessage.EscapeText(message);
 
-            var msg = _netManager.CreateNetMessage<MsgChatMessage>();
-            msg.Channel = ChatChannel.Local;
-            msg.Message = message;
-            msg.MessageWrap = Loc.GetString("chat-manager-entity-say-wrap-message",("entityName", Name: _entManager.GetComponent<MetaDataComponent>(source).EntityName));
-            msg.SenderEntity = source;
-            msg.HideChat = hideChat;
-            _netManager.ServerSendToMany(msg, clients);
+            var obfuscatedMessage = ObfuscateMessageReadability(message, 0.2f);
```

Comment:
You should check if there's any far clients before obfuscating as it may not be necessary.

Reply:
Done on commit #f9a5700